tests/foreman/cli/test_computeresource.py

```diff
@@ -347,24 +377,26 @@ def test_update_negative(self, options):
         @Assert: Compute Resource not updated
 
         """
-        comp_res = make_compute_resource()
-
-        result = ComputeResource.update(
-            dict({'name': comp_res['name']}, **options))
-
-        self.assertNotEqual(result.return_code, 0)
-        self.assertGreater(len(result.stderr), 0)
-
-        result = ComputeResource.info({'id': comp_res['id']})
-        self.assertEqual(result.return_code, 0)
-        # check attributes have not changed
-        self.assertEqual(result.stdout['name'], comp_res['name'])
-        options.pop('new-name', None)
-        for key in options.keys():
-            self.assertEqual(comp_res[key], result.stdout[key])
-
-    @data('true', 'false')
-    def test_set_console_password_v1(self, set_console_password):
+        for options in invalid_update_data():
+            with self.subTest(options):
+                comp_res = make_compute_resource()
```

Comment:
Move the object creation outside the loop and perform all updates inside the loop to speed up the test.